### ISSUE TITLE
Pagination is broken with "invalid memory or nil pointer dereference" in 0.123.0-DEV

### ISSUE THREAD

**Issue by TiGR** [gohugoio/hugo#11949]
I get error like this:

```
ERROR render of "page" failed: execute of template failed: template: _internal/pagination.html:21:17: executing "_internal/pagination.html" at <$page.Paginator.TotalPages>: error calling TotalPages: runtime error: invalid memory address or nil pointer dereference
```

And somewhat related issue is error like this:

```
executing "main" at <$paginate.Pages>: error calling Pages: runtime error: invalid memory address or nil pointer dereference
```

in code like this:

```
{{ $paginate := .Paginate $sliceOfPages 10 }}
{{ range $index, $page := $paginate.Pages }}
    {{ $page.Render "list_item" }}
{{ end }}
```

### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.123.0-DEV+extended linux/amd64 BuildDate=unknown
</pre>

### Does this issue reproduce with the latest release?
No, the latest stable release works well

**Comment by bep** [comment]
Thanks for this, much appreciated.

I don't think "pagination is broken" is a generally valid statement (we have lots of tests). Do you have an example site that shows this problem?

**Comment by jmooring** [comment]
@TiGR In which template does your code appear?

**Comment by TiGR** [comment]
In general, the idea is that we have structure like `/content/places/*/events/*/` and we're building a list of all places, but sorted by most recent event, and having each place only once. In order to achieve this we go through all events pages, sort them by some field. Then, as a final step we create a slice and then append all places pages in the order we need:

```html
{{ $places := slice }}
{{ range $events }}
  {{/* in this example .place is an item of (site.GetPage "places").Pages list, that is a Page */}}
  {{ $places = $places | append .place }}
{{ end }}
```

 And then we do the code I've mentioned above:

```html
<div class="places">
  {{ $paginate := .Paginate $places 10 }}
  {{ range $index, $page := $paginate.Pages }}
    {{ $page.Render "list_item" }}
  {{ end }}
</div>
<div class="pagination__wrap">
  {{- template "_internal/pagination.html" . -}}
</div>
```

With that code I get errors

**Comment by TiGR** [comment]
> Or, more specifically, is the pagination code called from a list page (home, section, taxonomy, or term), or from a single page?

It's a section page. But it's a tricky one, it's an empty sub-section with no pages inside (like `content/news/events/_index.md`), since we need to have it in a specific place of our website.

**Comment by jmooring** [comment]
Please rebuild hugo from HEAD at [6c3b6ba](https://github.com/gohugoio/hugo/commit/6c3b6ba3e6ccc220cbca9cc83fab78db0a78604e) and test again to see if the error message changes.

**Comment by TiGR** [comment]
Well. That's weird. The message has indeed changed. Now I get:

    ERROR Rebuild failed: render: failed to render pages: render of "page" failed: "...:25:33": execute of template failed at <.Paginate>: error calling Paginate: pagination not supported for pages of kind "page"

But the problem is, the page we're talking about is supposed to be a section [according to the documentation](https://gohugo.io/content-management/sections/#overview), and everything worked fine on previous versions. I have the following structure:

```
content/
├── news/
│   ├── events/
│   │   └── _index.md
│   └── _index.md
└── _index.md
```

And the only place were I have this template rendered is in `content/news/events/_index.md`. Which is supposed to be a section, but the hugo complains it is not.

Also, moving events directory to content top level fixes the issue, but I need it where it is now and I can't get it why it is not considered a section.

**Comment by TiGR** [comment]
Also, it might be important, we have a bunch of `_index.xx.html` translation files in `content/news/events` directory.

**Comment by bep** [comment]
@TiGR the devil is in the detail here, I guess, which is not possible for me to determine without seeing the full source.

But one of the improvements in this next version is a simplified logic in this area, to prepare for other content sources than files. So, the logic around handling of duplicates may have changed.

So, if you have something ala:

```
content/
├── news/
│   ├── events/
│   │   └── _index.md
│   │   └── index.md
│   └── _index.md
└── _index.md
```

Then that may explain it, but again, it's hard for me to guess without seeing the source.

That said, I appreciate your bug reports about this. I'm confident that we needed to make these changes, but I also understood that it could create some breakage in the wild.

**Comment by bep** [comment]
Can you try this with my latest commit in 1891d5e6b5575e3abb7e0f80c3fbce1670f9bd5e

**Comment by TiGR** [comment]
Well, that seems to be definately related with #11958, as I get the following:

```
ERROR render of "page" failed: "…/layouts/news/events.html:25:33": execute of template failed: template: news/events.html:25:33: executing "main" at <.Paginate>: error calling Paginate: pagination not supported for this page: kind: "page", path: "/news/events/_index.fi", file: "…/content/news/events/_index.fi.html"
ERROR render of "page" failed: "…/layouts/news/events.html:25:33": execute of template failed: template: news/events.html:25:33: executing "main" at <.Paginate>: error calling Paginate: pagination not supported for this page: kind: "page", path: "/news/events/_index.de", file: "…/content/news/events/_index.de.html"
ERROR render of "page" failed: "…/layouts/news/events.html:25:33": execute of template failed: template: news/events.html:25:33: executing "main" at <.Paginate>: error calling Paginate: pagination not supported for this page: kind: "page", path: "/news/events/_index.es", file: "…/content/news/events/_index.es.html"
ERROR render of "page" failed: "…/layouts/news/events.html:25:33": execute of template failed: template: news/events.html:25:33: executing "main" at <.Paginate>: error calling Paginate: pagination not supported for this page: kind: "page", path: "/news/events/_index.it", file: "…/content/news/events/_index.it.html"
Built in 71556 ms
Error: error building site: render: failed to render pages: render of "page" failed: "…/layouts/news/events.html:25:33": execute of template failed: template: news/events.html:25:33: executing "main" at <.Paginate>: error calling Paginate: pagination not supported for this page: kind: "page", path: "/news/events/_index.fr", file: "…/content/news/events/_index.fr.html"
```

Let me try building the entire website (this might take some time)

**Comment by TiGR** [comment]
Well, the entire website (without disabled languages builds well)

**Comment by bep** [comment]
I'm closing this, as this certainly is "another issue".

**Comment by jmooring** [comment]
I can reproduce the behavior if I erroneously add `kind = "page"` to the front matter of content/news/events/_index.md.
